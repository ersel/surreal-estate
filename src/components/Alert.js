/* eslint-disable import/no-extraneous-dependencies */
import styled from 'styled-components';
import PropTypes from 'prop-types';

const AlertBox = styled.div`
  width: 97%;
  height: auto;
  margin: 0 auto;
  margin-bottom: 20px;
  padding-left: 20px;

  ${({ isSuccess }) =>
    isSuccess
      ? `
  border: 1px solid #3997396b;
  background-color: #d4fbd466;
  `
      : `  border: 1px solid red;
  background-color: red;`}
`;

const AlertMessage = styled.p`
  margin: 10px 0;
  font-weight: 500;
  font-size: 1rem;
  text-align: left;
`;

const Alert = ({ message, isSuccess }) => {
  if (!message) return null;

  return (
    <AlertBox isSuccess={isSuccess}>
      <AlertMessage>{message}</AlertMessage>
    </AlertBox>
  );
};

Alert.defaultProps = {
  message: '',
};
Alert.propTypes = {
  message: PropTypes.string,
  isSuccess: PropTypes.bool.isRequired,
};

export default Alert;
