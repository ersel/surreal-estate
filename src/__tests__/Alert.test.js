import React from 'react';
import { render, screen } from '@testing-library/react';
import Alert from '../components/Alert';

describe('Alert', () => {
  test('it displays an error message', () => {
    const { asFragment } = render(<Alert message="Error!" isSuccess={false} />);
    expect(screen.getByText('Error!')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('it displays a success message', () => {
    const { asFragment } = render(
      <Alert message="Property added." isSuccess />
    );
    expect(screen.getByText('Property added.')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('it does not render when there is no alert message', () => {
    const { asFragment } = render(<Alert message="" isSuccess={false} />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });
});
