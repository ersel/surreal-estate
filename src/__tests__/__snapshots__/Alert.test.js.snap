// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert it displays a success message 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ pKYGM"
  >
    <p
      class="sc-gsTCUz cyCbzE"
    >
      Property added.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert it displays an error message 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ dcSyjK"
  >
    <p
      class="sc-gsTCUz cyCbzE"
    >
      Error!
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert it does not render when there is no alert message 1`] = `<DocumentFragment />`;
